src/lib/routers/history.ts

```diff
@@ -88,50 +138,39 @@ class BrowserHistory {
   }
 
   /**
-   * This methods read the URL and returns a syncable UI search state.
-   * @return {object} the equivalent to what is store in the URL as an object
+   * Sets a callback on the `onpopstate` event of the history API of the current page.
+   * It enables the URL sync to keep track of the changes.
    */
-  read() {
-    return this.parseURL({ qsModule: qs, location: window.location });
-  }
-
-  /**
-   * This methods sets a callback on the `onpopstate` event of the history API
-   * of the current page. This way, the URL sync can keep track of the changes.
-   * @param {function(object)} cb the callback that will receive the latest routeState.
-   * It is called when the URL is updated.
-   * @returns {undefined}
-   */
-  onUpdate(cb) {
+  public onUpdate(callback: (routeState: UiState) => void): void {
```

Comment:
How can routeState === uiState?

Reply:
`UiState` is rather generic but you're right, let's create another type.